From df6afce2d82e0a2c32938545bc132c1a5ca302fa Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Wed, 02 Feb 2022 20:14:29 +0200
Subject: [PATCH] [SW-67835] greco: add missing pcie gen3 and gen4 configuration

This patch adds missing EQ REDO pcie configuration for gen3 and
gen4, as shared by pcie sival team.

Change-Id: I6a8eae957eef59bb758d48b5f217e80b459b99f3
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index c732c30..a0e2a08 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -484,8 +484,13 @@
 
 	preboot_pcie_device_id_vendor_id_set();
 
-	/* Disable the EQ REDO */
+	/* Disable the EQ REDO Gen3 */
 	regmap_writel(0x00402801, mmPCIE_DBI_GEN3_RELATED_OFF);
+	regmap_writel(0x04027041, mmPCIE_DBI_GEN3_EQ_CONTROL_OFF);
+
+	/* Disable the EQ REDO Gen4 */
+	regmap_writel(0x01402801, mmPCIE_DBI_GEN3_RELATED_OFF);
+	regmap_writel(0x04027041, mmPCIE_DBI_GEN3_EQ_CONTROL_OFF);
 
 	start = timestamp_get_us();
 
